vars/mtaBuild.groovy

```diff
@@ -1,189 +1,11 @@
-            def applicationName = configuration.applicationName
-
-            if (!fileExists(mtaYamlName)) {
-                if (!applicationName) {
-                    error "'${mtaYamlName}' not found in project sources and 'applicationName' not provided as parameter - cannot generate '${mtaYamlName}' file."
-                } else {
-                    echo "[INFO] '${mtaYamlName}' file not found in project sources, but application name provided as parameter - generating '${mtaYamlName}' file."
-                    MtaUtils mtaUtils = new MtaUtils(this)
-                    mtaUtils.generateMtaDescriptorFromPackageJson("package.json", mtaYamlName, applicationName)
-                }
-            } else {
-                echo "[INFO] '${mtaYamlName}' file found in project sources."
-            }
-
-            //[Q]: Why not yaml.dump()? [A]: This reformats the whole file.
-            sh "sed -ie \"s/\\\${timestamp}/`date +%Y%m%d%H%M%S`/g\" \"${mtaYamlName}\""
-
-            def mtaCall
-            def options = []
-
-            String mtarName = configuration.mtarName?.trim()
-            if (!mtarName) {
-                def mtaId = getMtaId(mtaYamlName)
-                mtarName = "${mtaId}.mtar"
-            }
-            options.push("--mtar ${mtarName}")
-
-            switch(configuration.mtaBuildTool) {
-                case 'classic':
-                    // If it is not configured, it is expected on the PATH
-                    def mtaJar = configuration.mtaJarLocation ?: 'mta.jar'
-                    options.push("--build-target=${configuration.buildTarget}")
-                    if (configuration.extension) options.push("--extension=${configuration.extension}")
-                    mtaCall = "java -jar ${mtaJar} ${options.join(' ')} build"
-                    break
-                case 'cloudMbt':
-                    options.push("--platform ${configuration.platform}")
-                    options.push("--target ./")
-                    if (configuration.extension) options.push("--extensions=${configuration.extension}")
-                    mtaCall = "mbt build ${options.join(' ')}"
-                    break
-                default:
-                    error "[ERROR][${STEP_NAME}] MTA build tool '${configuration.mtaBuildTool}' not supported!"
-            }
-
-            echo "[INFO] Executing mta build call: '${mtaCall}'."
-
-            //[Q]: Why extending the path? [A]: To be sure e.g. grunt can be found
-            //[Q]: Why escaping \$PATH ? [A]: We want to extend the PATH variable in e.g. the container and not substituting it with the Jenkins environment when using ${PATH}
-            sh """#!/bin/bash
-            export PATH=./node_modules/.bin:\$PATH
-            $mtaCall
-            """
-
-            script?.commonPipelineEnvironment?.setMtarFilePath("${mtarName}")
-        }
-    }
-}
+    handlePipelineStepErrors (stepName: STEP_NAME, stepParameters: parameters) {
```

Comment:
this is not required since `piperExecuteBin` has a `hanbdlePipelineStepErrors` closure with the passed step name, in this case `mtaBuild`

Reply:
Done.

To my understanding it is a bug to report about the step name passed to `piperExecuteBin` rather the the step name of `piperExecuteBin` since that makes troubleshooting harder. But that is another story. We will also have steps where more code remains inside the stub (e.g. locking) hence it is not in general superfluous to have `handlePipelineStepErrors` inside the stubs. But since here almost nothing remains we can also drop it.